discord/embeds.py

```diff
@@ -0,0 +1,80 @@
+import datetime
+from typing import Union, Dict
+
+__all__ = "Embed"
+
+
+class Embed:
+    def __init__(self, *, title: str = None,
+                 description: str = None,
+                 color: int = 0x000000,
+                 timestamp: Union[datetime.datetime, str] = None,
+                 url:str = None,
+                 type: str = "rich"):
+        self.title = str(title) if title else None
+        self.description = str(description) if description else None
+        self.timestamp = str(timestamp) if timestamp else None
+        self.color = color
+        self.url = url
+        self.fields = []
+
+    def _to_dict(self):
+        return {key: value for key, value in self.__dict__.items() if value is not None}
+
+    def add_field(self, *, name, value, inline = True):
```

Comment:
Adding a way to set the index of the field would be cool. Set it to typing.Optional and a default value of None, and it will append to the end. If an index is given try to insert on the index, or raise an indexerror if there aren't enough fields yet to insert it here.